core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -795,7 +796,8 @@ private[spark] object JsonProtocol {
       case `taskKilled` => TaskKilled
       case `executorLostFailure` =>
         val executorId = Utils.jsonOption(json \ "Executor ID").map(_.extract[String])
-        ExecutorLostFailure(executorId.getOrElse("Unknown"))
+        val reason = Utils.jsonOption(json \ "Loss Reason").map(_.extract[String])
+        ExecutorLostFailure(executorId.getOrElse("Unknown"), reason.getOrElse("Unknown"))
```

Comment:
Ah I see, misread the code. I'd still use an `Option[String]` for the reason instead of defaulting to `"Unknown"`.